R/net.mod.infection.R

```diff
@@ -331,6 +331,121 @@ discord_edgelist <- function(dat, at, network = 1, infstat = "i", include.networ
   return(del)
 }
 
+#' @title Get Discordant Edgelist Based on Specified Status Variable
+#'
+#' @description This function returns a `data.frame` with a discordant
+#'              edgelist, defined as the set of edges for which the status attribute
+#'              of interest is discordant between the two partners.
+#'
+#' @inheritParams get_cumulative_edgelists_df
+#' @param status.attr The name of the status attribute of interest.
+#' @param head.status The value(s) of `status.attr` for which to look for the head of the edge.
+#'        Can be a single value or a vector.
+#' @param tail.status  The value(s) of `status.attr` for which to look for the tail of the edge.
+#'        Can be a single value or a vector.
+#'
+#' @details
+#' This is a generalized version of the `discord_edgelist` function.
+#' It creates an edgelist of current partnerships in which the status attribute
+#' of interest (as specified by the parameter `status.attr`) of one partner matches
+#' the value (or one of the values) of the `head.status` parameter while the
+#' corresponding status attribute of the other partner matches the value (or
+#' one of the values) of the `tail.status` parameter.
+#'
+#' @return
+#' A `data.frame` with the following columns:
+#'  * `head`: Positional ID of the head node.
+#'  * `tail`: Positional ID of the tail node.
+#'  * `head_status`: Status of the head node.
+#'  * `tail_status`: Status of the tail node.
+#'  * `network`: The numerical index of the network on which the partnership is located.
+#'
+#' @seealso \code{\link{discord_edgelist}}
+#'
+#' @export
+#' @keywords netMod internal
+#'
+get_discordant_edgelist <- function(dat, status.attr, head.status,
+                                    tail.status, networks = NULL) {
+
+  if (length(intersect(head.status, tail.status)) > 0 &&
+        get_current_timestep(dat) == (get_control(dat, "start") + 1)) {
```

Comment:
I think the  time condition should be first here.
The `&&` operator is short circuited which means that it will only perform the `intersect` call if at time `start + 1`. And in general, set operation (`intersect`, `setdiff`, etc) are more computationally intensive compared to getters